examples/rvc-app/rvc-common/include/rvc-mode-delegates.h

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/util/af.h>
+#include <app/util/config.h>
+#include <cstring>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class RvcDevice;
+
+typedef void (RvcDevice::*HandleChangeToModeA)(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response);
+
+namespace RvcRunMode {
+
+const uint8_t ModeIdle     = 0;
+const uint8_t ModeCleaning = 1;
+const uint8_t ModeMapping  = 2;
+
+/// This is an application level delegate to handle RvcRun commands according to the specific business logic.
+class RvcRunModeDelegate : public ModeBase::Delegate
+{
+private:
+    using ModeTagStructType               = detail::Structs::ModeTagStruct::Type;
+    ModeTagStructType ModeTagsIdle[1]     = { { .value = to_underlying(ModeTag::kIdle) } };
+    ModeTagStructType ModeTagsCleaning[1] = { { .value = to_underlying(ModeTag::kCleaning) } };
+
+    const detail::Structs::ModeOptionStruct::Type kModeOptions[3] = {
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Idle"),
+                                                 .mode     = ModeIdle,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(ModeTagsIdle) },
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Cleaning"),
+                                                 .mode     = ModeCleaning,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(ModeTagsCleaning) },
+        detail::Structs::ModeOptionStruct::Type{
+            .label    = CharSpan::fromCharString("Mapping"),
+            .mode     = ModeMapping,
+            .modeTags = DataModel::List<const ModeTagStructType>(ModeTagsIdle) }, // todo set to no mode tags
```

Comment:
Perhaps use a manufacturer specific value, so we can test that too.

Reply:
Used `TestVendor1` and `0x8001`.